### DISCUSSION TITLE
"Base" dev environment with Okteto

### DISCUSSION THREAD

**Discussion by AgustinRamiroDiaz** [okteto/okteto#3049] [Ideas]
# Motivation
Having the hustles of working with WSL and the benefits of Okteto with remote dev environment, I think it would be awesome to have a way to spin up a "base" dev environment with all the tooling necessary to work on different dev environments.

It would be something like [vscode remote development](https://code.visualstudio.com/learn/develop-cloud/overview), [github codespaces](https://github.com/features/codespaces) and [jetbrains remote development](https://www.jetbrains.com/remote-development/). I feel that the okteto extension for vscode does something similar, by removing the processing from the local machine to the remote one (vscode extensions are heavy), but we are still confined to one development environment

# Some ideas
I imagine it like this:
- Only have okteto installed on local machine
- Having some manifest/dotfiles which describe the base development environment. Something similar to the [`.devcontainer`](https://code.visualstudio.com/docs/remote/create-dev-container) of VSCode remote development
  - This should describe all the common meta tooling around development, like `kubectl` or `helm`, to have them available inside the "base" dev environment
  - File sync, for example to import credentials to use this tools 
- This "base" devenv should have okteto installed to be able to `okteto up` into other dev environments
- This "base" devenv could live in its own namespace or even cluster, as it won't 

In summary, it'd be a mix of github codespaces with okteto.   

# Product thoughts
Okteto could leave this out of scope, as there's are similar way to accomplish this with a custom dev image + vscode extension. But I think it's something important to bear in mind when thinking of remote dev environment experience as a whole
Do we need to have all this stuff installed on our local machines?
Can we reduce the dependencies from our laptops?
